services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py

```diff
@@ -0,0 +1,162 @@
+import logging
+from decimal import Decimal
+from typing import cast
+
+import sqlalchemy as sa
+from models_library.api_schemas_resource_usage_tracker.credit_transactions import (
+    WalletTotalCredits,
+)
+from models_library.products import ProductName
+from models_library.resource_tracker import CreditTransactionId, CreditTransactionStatus
+from models_library.wallets import WalletID
+from simcore_postgres_database.models.resource_tracker_credit_transactions import (
+    resource_tracker_credit_transactions,
+)
+from simcore_postgres_database.utils_repos import transaction_context
+from sqlalchemy.ext.asyncio import AsyncConnection, AsyncEngine
+
+from ....exceptions.errors import CreditTransactionNotCreatedDBError
+from ....models.credit_transactions import (
+    CreditTransactionCreate,
+    CreditTransactionCreditsAndStatusUpdate,
+    CreditTransactionCreditsUpdate,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+async def create_credit_transaction(
+    engine: AsyncEngine,
+    connection: AsyncConnection | None = None,
+    *,
+    data: CreditTransactionCreate
+) -> CreditTransactionId:
+    async with transaction_context(engine, connection) as conn:
+        insert_stmt = (
+            resource_tracker_credit_transactions.insert()
+            .values(
+                product_name=data.product_name,
+                wallet_id=data.wallet_id,
+                wallet_name=data.wallet_name,
+                pricing_plan_id=data.pricing_plan_id,
+                pricing_unit_id=data.pricing_unit_id,
+                pricing_unit_cost_id=data.pricing_unit_cost_id,
+                user_id=data.user_id,
+                user_email=data.user_email,
+                osparc_credits=data.osparc_credits,
+                transaction_status=data.transaction_status,
+                transaction_classification=data.transaction_classification,
+                service_run_id=data.service_run_id,
+                payment_transaction_id=data.payment_transaction_id,
+                created=data.created_at,
+                last_heartbeat_at=data.last_heartbeat_at,
+                modified=sa.func.now(),
+            )
+            .returning(resource_tracker_credit_transactions.c.transaction_id)
+        )
+        result = await conn.execute(insert_stmt)
+    row = result.first()
```

Comment:
I will point out that you are using the non blocking version of the cursor here. You need to await this.